### PR TITLE
Holidays usability cleaning + allocation per department

### PR DESCRIPTION
**Pull request by rim-odoo**
- https://www.odoo.com/web?debug#id=31292&view_type=form&model=project.task&action=327&menu_id=4720

### CODE REVIEW

File: addons/hr_holidays/models/hr_holidays.py [hunk 1]
```diff
@@ -375,6 +383,10 @@ def unlink(self):
             raise UserError(_('You cannot delete a leave which is in %s state.') % (holiday.state,))
         return super(Holidays, self).unlink()
 
+    @api.multi
+    def copy_data(self, default=None):
```
Comment (tde-banana-odoo) [inline addons/hr_holidays/models/hr_holidays.py]: Why ?

File: addons/hr_holidays/models/hr_holidays.py [hunk 2]
```diff
@@ -442,7 +454,7 @@ def action_approve(self):
         return True
 
     @api.multi
-    def _prepare_create_by_category(self, employee):
+    def _prepare_create_by_batch(self, employee):
```
Comment (tde-banana-odoo) [inline addons/hr_holidays/models/hr_holidays.py]: The new name is not clearer than the old one ...

File: addons/hr_holidays/models/hr_holidays.py [hunk 3]
```diff
@@ -481,10 +493,10 @@ def action_validate(self):
                 holiday.write({'first_approver_id': current_employee.id})
             if holiday.holiday_type == 'employee' and holiday.type == 'remove':
                 holiday._validate_leave_request()
-            elif holiday.holiday_type == 'category':
+            elif holiday.holiday_type in ['category', 'department']:
                 leaves = self.env['hr.holidays']
-                for employee in holiday.category_id.employee_ids:
-                    values = holiday._prepare_create_by_category(employee)
+                for employee in holiday.category_id.employee_ids if holiday.holiday_type == 'category' else holiday.department_id.member_ids:
```
Comment (tde-banana-odoo) [inline addons/hr_holidays/models/hr_holidays.py]: I would 1/ compute employees 2/ perform the for. HAving the 2 in one line does not help understanding the code.

File: addons/hr_holidays/views/hr_holidays_views.xml [hunk 4]
```diff
@@ -153,6 +148,22 @@
             </field>
         </record>
 
+        <record id="view_holiday_my_allocation_tree" model="ir.ui.view">
```
Comment (tde-banana-odoo) [inline addons/hr_holidays/views/hr_holidays_views.xml]: Just XML-Iding Things: hr_holidays_view_list_blahblah

File: addons/hr_holidays/models/hr_holidays.py (hunk 1 above)
Comment (rim-odoo) [reply addons/hr_holidays/models/hr_holidays.py]: Was a request of the HR, else people who have rights duplicate their approved holidays and then change the date...

File: addons/hr_holidays/models/hr_holidays.py (hunk 2 above)
Comment (rim-odoo) [reply addons/hr_holidays/models/hr_holidays.py]: /agree

File: addons/hr_holidays/views/hr_holidays_views.xml (hunk 4 above)
Comment (rim-odoo) [reply addons/hr_holidays/views/hr_holidays_views.xml]: :+1:

File: addons/hr_holidays/models/hr_holidays.py (hunk 3 above)
Comment (rim-odoo) [reply addons/hr_holidays/models/hr_holidays.py]: hagrid

**Pull request merged**